theforeman/foreman-packaging
# Fixes #28939 - change-hostname: update DNS records when changing hostname

When applicable, this change updates DNS records using `nsupdate` during the `[scenario]-change-hostname` script.

To test on a sat-deploy box:
* make sure DNS is managed by satellite (run the satellite-provisioning playbook)
* View the contents of `/var/named/dynamic`

```
sudo su
cd /var/named/dynamic
ls -l
```
You should see a DNS zone file named for your zone, such as `db.example.com`.  Inspect the contents and note that they contain the current satellite hostname in the SOA, A, and NS records:
```
[root@sat-6-7-qa-rhel7 dynamic]# cat db.example.com
$TTL 10800
@ IN SOA sat-6-7-qa-rhel7.redhatlaptop.example.com. root.example.com. (
	1	;Serial
	86400	;Refresh
	3600	;Retry
	604800	;Expire
	3600	;Negative caching TTL
)

@ IN NS sat-6-7-qa-rhel7.redhatlaptop.example.com.

sat-6-7-qa-rhel7.redhatlaptop.example.com. IN A 192.168.73.1
```
The reverse zone file (named similar to `db.73.168.192.in-addr.arpa`) will also contain records referring to the current hostname.

* Set permissions on `hostname-change.rb`:
```
cd /usr/share/katello
chmod a+w hostname-change.rb
```

* On your local machine in the `sat-deploy` directory, run
```
vagrant ssh-config [box_name]
```
* Take note of the IdentityFile and HostName values (HostName will actually be an IP)

* Copy the new `hostname-change.rb` from this PR to the sat-deploy box using `scp`:
```
scp -i $IDENTITY_FILE $LOCAL_HOSTNAME_CHANGE.RB vagrant@$HOSTNAME:/usr/share/katello/hostname-change.rb
```
* Verify using `cat` or `vim` that the `hostname-change.rb` contents successfully copied over
* Repeat the above with `helper.rb`

* Run
```
vim /etc/foreman-installer/scenarios.d/satellite-answers.yaml
```
* Edit the `dns_provider` value (under `foreman_proxy:`) to read `nsupdate`

* Run 
```
satellite-change-hostname -u admin -p changeme newhost.mydomain.example.com
```
* View new output
```
gathering DNS data...
updating DNS records:
forward...
reverse...
updating dynamic zone files...
DNS records updated
```

* Verify that the contents of the zone files `/var/named/db.$ZONE` now reflect the new hostname

This should allow PXE discovery to continue to work properly.
